### ISSUE TITLE
Clarification on mfxAISuperResolutionMode Support and Implementation in libvpl

### ISSUE THREAD

**Issue by quamt** [intel/libvpl#156]
Hello,I am seeking to better understand the mfxAISuperResolutionMode feature within Intel’s libvpl. 

I would appreciate some clarification on the following points:

Driver and Architecture Support:

Which Intel GPU architectures and driver versions support the mfxAISuperResolutionMode feature? 
Is there a resource or changelog available that details when this feature was introduced or enhanced across different platforms in libvpl?

Understanding Application Integration:

I understand that the libvpl dispatcher links the application with the appropriate runtime libraries. Could you explain the recommended process for initializing and checking the availability of mfxAISuperResolutionMode within an application?

Additionally, what considerations should be made if this feature is not supported on certain hardware or driver versions? 

Are there suggested fallback mechanisms or best practices in such cases?

Thank you for your guidance. 
Any documentation or resources that could further my understanding of this feature and its integration would be greatly appreciated.

**Comment by jonrecker** [comment]
Thank you for your interest in this feature. Some documentation and code snippets are available in the VPP [programming guide](https://intel.github.io/libvpl/latest/programming_guide/VPL_prg_vpp.html#ai-powered-video-processing) and the API [reference](https://intel.github.io/libvpl/latest/API_ref/VPL_structs_vpp.html#mfxextvppaisuperresolution).

@FurongZhang could you please provide some additional info re. the above questions?

**Comment by FurongZhang** [comment]
1. VPL API: We added mfxAISuperResolutionMode in VPL v2.11 as experimental API, v2.13 move to production. 
2. Driver version: 
a. The latest driver supports AI power super resolution (still experimental API) https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html. The officially released driver in 2024 Q2/Q3 drivers can support this feature. 
b. This feature is only supported on Windows, NO on Linux. 
c. This feature has been supported on TigerLake, Metorlake, LunarLake. A lot of platforms are compatible. 
d. As the programming guide shows, use query or init to check the capability of super resolution. If failed, recommend to fall back to the VPL hardware scaling path(no-AI).

**Comment by quamt** [comment]
@FurongZhang
Thank you for your explanation.

Does this also work on Intel Arc GPUs?

**Comment by FurongZhang** [comment]
Yes, I works on Intel Arc GPUs. I assume you are talking about DG2 series.

**Comment by quamt** [comment]
> Yes, I works on Intel Arc GPUs. I assume you are talking about DG2 series.

Yes, correct the DG2 series. 
Arc A750 for example.

I tested to check it with the 64-bit version of QSVENc from the git hub repository here:
`https://github.com/rigaya/QSVEnc/releases` 7.69

Tested with latest driver that I found:
`GPU: Intel Arc A750 Graphics (448EU) 300-2400MHz (32.0.101.6078)`

Here it shows as X for not supported
`AI Super Res          x`


Supported VPP features:

```
Resize                o
AI Super Res          x
Deinterlace           o
Scaling Quality       o
Denoise               o
Denoise2              o
Mctf                  x
Rotate                o
Mirror                o
Detail Enhancement    o
Proc Amp.             o
Image Stabilization   x
Perceptual Pre Enc    o
Video Signal Info     o
FPS Conversion        o
FPS Conversion (Adv.) o
AI Frame Interp       o
```

Thats why I was curious on how the detection feature works.

**Comment by FurongZhang** [comment]
What is the input / output format you are using?

**Comment by quamt** [comment]
> What is the input / output format you are using?

I didn't try to encode something yet, with that setting enabled in the parameters.
The setting listed to enable is:
`--vpp-resize mfx-ai-superres`

I usually encode with hevc 10-bit or avc as input and also as output.

**Comment by FurongZhang** [comment]
I did not mean encoding. I mean VPP input / output format.

**Comment by quamt** [comment]
> I did not mean encoding. I mean VPP input / output format.

Can you please elaborate?

**Comment by FurongZhang** [comment]
The input and output of VPP for AI super resolution is NV12. Other formats might not be supported.

**Comment by quamt** [comment]
> The input and output of VPP for AI super resolution is NV12. Other formats might not be supported.

Thank you for the clarification.
I need to investigate and get back on this.

**Comment by quamt** [comment]
@FurongZhang 

It uses NV12.

QSVEnc 7.70 updates to latest libvpl to support API 2.13.

6078 driver support is up to API 2.12?
Could there be a driver problem?

**Comment by tletnes** [comment]
@FurongZhang Can you respond?

**Comment by roycezjq** [comment]
@wizziwig @tletnes  Hi do you mind share where I can find the sample_vpp.exe? I can't find it in this repo or the oneVPL document site. Thanks

**Comment by FurongZhang** [comment]
https://github.com/intel/libvpl-tools/tree/main/tools/legacy/sample_vpp
